R/reacttools.R

```diff
@@ -0,0 +1,91 @@
+isUpper <- function(s) {
+  grepl("^[[:upper:]]+$", s)
+}
+
+#' Create a React component
+#'
+#' @param name Name of the React component, which must start with an upper-case
+#'   character.
+#' @param varArgs Attributes and children of the element to pass along to
+#'   \code{\link[htmltools]{tag}} as \code{varArgs}.
+#'
+#' @return An htmltools \code{\link[htmltools]{tag}} object
+#' @export
+#'
+#' @examples
+#' component("ParentComponent",
+#'   list(
+#'     x = 1,
+#'     y = 2,
+#'     component("ChildComponent"),
+#'     component("OtherChildComponent")
+#'   )
+#' )
+component <- function(name, varArgs = list()) {
+  if (length(name) == 0 || !isUpper(substring(name, 1, 1))) {
+    stop("Component name must be specified and start with an upper case character")
+  }
+  htmltools::tag(name, varArgs)
+}
```

Comment:
Might want to add a class or a subclass here? As a package author, I might want to know if a user is supplying a component versus a tag